src/utils/arappUtils.ts

```diff
@@ -25,28 +25,34 @@ export function readArappIfExists(): AragonAppJson | undefined {
 
 /**
  * Returns app ens name.
+ * @param network "mainnet"
  * @return "voting.open.aragonpm.eth"
  */
-export function getAppEnsName(): string {
+export function getAppEnsName(network = 'default'): string {
   const arapp = readArapp()
 
-  const defaultEnvironment = arapp.environments.default
-  if (!defaultEnvironment) {
-    throw new BuidlerPluginError('Default environemnt not found in arapp.json')
+  const environment: AragonEnvironment | undefined =
+    arapp.environments[network] || arapp.environments['default']
+  if (!environment) {
+    throw new BuidlerPluginError(`Default environemnt not found in arapp.json.`)
   }
 
-  const appName = defaultEnvironment.appName
-  if (!appName) throw Error('No appName found in environment')
+  const appEnsName = (environment || {}).appName
+  if (!appEnsName)
```

Comment:
We check the `appEnsName` early and we assure this values exist later on logic like artifact generation.